src/pages/profile/Profile.jsx

```diff
@@ -0,0 +1,216 @@
+import './Profile.scss';
+
+// Reac- bootstrap components 
+import Tab from 'react-bootstrap/Tab';
+import Tabs from 'react-bootstrap/Tabs';
+
+// react icons
+import { AiFillEdit } from 'react-icons/ai';
+import { MdFileUpload } from 'react-icons/md';
+
+import AddImg from '../../imgs/addAvatar.png';
+
+// image upload function
+// in second sprint a good idea would be to trasfer all firebase funtionality every file to firebase.js
+
+const Profile = () => {
+
+    // User context
+    const { currentUser } = useContext(AuthContext);
+
+    /*
+    User profile functionality
+  */
+
+    // default avatar png       
+    const [avatarUrl, setAvatarUrl] = useState(AddImg);
+    const [photoBinary, setPhotoBinary] = useState(null);
+    const [loading, setLoading] = useState(false);
+
+    // username state
+    const [userName, setUserName] = useState(currentUser.displayName);
+
+    const [changeData, setChangeData] = useState({
+        nickname: ""
+        //  changed data object
+    })
+
+    useEffect(() => {
+
+        if (currentUser?.photoURL) {
+            setAvatarUrl(currentUser.photoURL);
+        }
+        setUserName(currentUser.displayName);
+
+    }, [currentUser.photoURL, currentUser.displayName])
+
+
+    const handleChange = (e) => {
+        if (e.target.files[0]) {
+            setPhotoBinary(e.target.files[0]);
+        }
+    }
+
+    const handleUpload = () => {
+        handleAvatarUpload(photoBinary);
+        setPhotoBinary();
+    }
+
+    const handleAvatarUpload = async (file) => {
+        const fileRef = ref(storage, currentUser.uid + ".png");
+        setLoading(true);
+
+        const snapshot = uploadBytesResumable(fileRef, file)
+
+        // accessing the url of the photo
+        const photoURL = await getDownloadURL(fileRef)
+
+        updateProfile(currentUser, { photoURL });
+
+        setLoading(false);
+        alert("File was uploaded");
+
+    }
+
+    // nickname input functionality
+    const handleNameChange = (event) => {
+        const { name, value } = event.target;
+
+        setChangeData({
+            ...changeData,
+            [name]: value,
+        })
+        console.log(changeData.nickname)
+        // Still to add error validation
+    }
+
+    //Need to review the rerender of the name
+
+    const handleNameChangeSubmit = (event) => {
+        event.preventDefault();
+        updateProfile(currentUser, { displayName: changeData.nickname });
+        if (event.target[0].value) {
+            event.target[0].value = '';
+        }
+        console.log(currentUser)
+    }
+
+    /*
+    User auth functionality
+    */
+
+    const [emailAuth, setEmailAuth] = useState('');
+    const [errorEmailAuth, setErrorEmailAuth] = useState('');
+
+    const validateEmailField = (value) => {
+
+        if (!value.trim()) {
+            return 'Email is required';
+        } else if (!/\S+@\S+\.\S+/.test(value)) {
+            return 'Email is invalid';
+        }
+        return '';
+    };
+
+
```

Comment:
Already commented before. User shouldn't have an ability to change the only contact method in existence.